future-apps/payroll/README.md

```diff
@@ -1,71 +1,185 @@
+
+Add ETH and DAI as allowed tokens for salary withdrawl:
+
 ```
-payroll.setExchangeRate(address token, uint256 denominationExchangeRate)
+payroll.addAllowedToken(0x0000000000000000000000000000000000000000); // ETH
+payroll.addAllowedToken(0x89d24a6b4ccb1b6faa2625fe562bdd9a23260359); // DAI
 ```
-Set the exchange rate for an allowed token against the Payroll denomination token. It needs `ORACLE_ROLE`.
 
-#### Add employee
-Three options can be used:
+4. **Add employee**
+
+Add an employee starting on Jan. 1st, 2020, with 80,000 DAI salary (note the salary is in seconds, and already adjusted for 18 decimals):
+
 ```
-payroll.addEmployee(address accountAddress, uint256 initialYearlyDenominationSalary, string role, uint256 startDate)
+payroll.addEmployee(0xabc...def, 2535047025122317, 1577836800, "employee");
 ```
-Add employee to the organization. Start date is used as the initial payment day. If it's not provided, the date of the transaction will be used. It needs `ADD_EMPLOYEE_ROLE`.
 
-#### Modify employee salary
+Assume for future examples this employee has an employee ID of `1`.
+
+5. **Modify employee salary**
+
+After some time, give the employee a raise by changing their salary to 90,000 DAI:
+
 ```
-payroll.setEmployeeSalary(uint128 employeeId, uint256 yearlyDenominationSalary)
+payroll.setEmployeeSalary(1, 2851927903262606);
 ```
-It needs `ADD_EMPLOYEE_ROLE`.
 
-#### Remove employee
+6. **Terminate employee**
+
+Finally, remove the employee from payroll on Jan. 1, 2022:
+
 ```
-payroll.removeEmployee(uint128 employeeId)
+payroll.terminateEmployee(1, 1640995200);
 ```
-Remove employee from organization. The owed up to current date salary will be transferred to the employee. It needs `REMOVE_EMPLOYEE_ROLE`.
 
-#### Determine allocation
+### Employee
+
+1. **Determine allocation**
+
 ```
 payroll.determineAllocation(address[] tokens, uint8[] distribution)
 ```
+
 Employees can set the proportion of every allowed token that want to be used for their salary payment. Distribution values are expressed as a ratio to 100.
 
-#### Request payroll
+2. **Request payroll**
+
 ```
 payroll.payday()
 ```
+
 Employees can request payroll whenever they want and the proportional amount of their anual salary since the last request (or since the start date if it's the first one) will be transferred.
 
-#### Change account address
+3. **Change account address**
+
 ```
 payroll.changeAddressByEmployee(address newAddress)
 ```
+
 Employees can change their own address.
 
-### Limitations
+4. **Claim final paycheck after termination**
+
+```
+payroll.payday()
+```
+
+## :warning: Gotchas to be aware of / potentially dangerous limitations
+
+### Requirements For allowed and denomination tokens
+
+Due to the limits of mathematical precision, any token used as either the denonination token or an allowed token must have a "reasonable" exchange rate against all other allowed or denomination tokens and this exchange rate must be fetchable from the price feed.
```

Comment:
```suggestion
Due to the limits of mathematical precision, any token used as either the denomination token or an allowed token must have a "reasonable" exchange rate against all other allowed or denomination tokens and this exchange rate must be fetchable from the price feed.
```

Reply:
I think we only care there is an exchange rate agains the denomination token right? I mean, we don't do any calculations between allowed tokens.